packages/core/src/common/resource.ts

```diff
@@ -30,12 +30,13 @@ export interface Resource extends Disposable {
     saveContents?(content: string, options?: { encoding?: string }): Promise<void>;
     saveContentChanges?(changes: TextDocumentContentChangeEvent[], options?: { encoding?: string }): Promise<void>;
     readonly onDidChangeContents?: Event<void>;
+    guessEncoding?(): Promise<string | undefined>
 }
 export namespace Resource {
     export interface SaveContext {
         content: string
         changes?: TextDocumentContentChangeEvent[]
-        options?: { encoding?: string }
+        options?: { encoding?: string, overwriteEncoding?: string }
```

Comment:
Could you elaborate why do we need `overwriteEncoding` and cannot just use `encoding`?

Reply:
So if you want to overwrite the encoding of a file, you will need 1. the original encoding of the file; 2. the new encoding.

To be clearer, when only 'encoding' parameter is presented, it means one wants to read and write a file in that encoding. And when 'overwriteEncoding' is explicitly used, it means that the file will be read in 'encoding', and be written in 'overwriteEncoding'.

Reply:
Should not we just call once `setContent` with old content and new encoding to overwrite the original content and then use new encoding for consequent `setContent` and `updateContent` methods? then we don't need `overwriteEncoding`?

Reply:
Please look at "node-filesystem.ts/updateContent()". If "contentChanges.length === 0", this method returns directly, and no encoding changes will be applied.